Lib/fontTools/varLib/__init__.py

```diff
@@ -601,6 +601,30 @@ def _add_MVAR(font, masterModel, master_ttfs, axisTags):
 		mvar.ValueRecord = sorted(records, key=lambda r: r.ValueTag)
 
 
+def _add_BASE(font, masterModel, master_ttfs, axisTags):
+
+	log.info("Generating BASE")
+
+	merger = VariationMerger(masterModel, axisTags, font)
+	merger.mergeTables(font, master_ttfs, ['BASE'])
+	store = merger.store_builder.finish()
+
+	if not store.VarData:
+		return
+	try:
+		BASE = font['BASE'].table
+		assert BASE.Version == 0x00010000
+	except KeyError:
```

Comment:
I think you can unconditionally expect that the `font` has a `'BASE'` table instead of handling `KeyError`, because calling `_add_BASE` method only makes sense in so far as at least the neutral master (and possibly also other masters) contain a `BASE` table (see my `if 'BASE' in vf` suggestion in the another review comment).

The neutral master is duplicated and becomes the basis where the rest of the VF tables are added. If the neutral doesn't contain `BASE`, then the VarStore you get from the VarStore builder ought to be empty, thus this `if` branch will never be hit, as you return early when an empty VarStore is found.

So just assume `BASE` is going to be in the `font`; the KeyError is avoided by not calling `_add_BASE` when neutral master  doesn't contain `BASE` in the first place

Reply:
Ok, removed the whole try/except since it's redundant and unnecessary.